crates/forge/bin/cmd/test/mod.rs

```diff
@@ -392,31 +416,10 @@ impl TestArgs {
         let mut outcome = TestOutcome::empty(self.allow_failure);
 
         let mut any_test_failed = false;
-        for (contract_name, mut suite_result) in rx {
+        for (contract_name, suite_result) in rx {
             let tests = &suite_result.test_results;
 
-            // Set up trace identifiers.
-            let known_contracts = &suite_result.known_contracts;
-            let mut identifier = TraceIdentifiers::new().with_local(known_contracts);
-
-            // Avoid using etherscan for gas report as we decode more traces and this will be
-            // expensive.
-            if !self.gas_report {
-                identifier = identifier.with_etherscan(&config, remote_chain_id)?;
-            }
-
-            // Build the trace decoder.
-            let mut builder = CallTraceDecoderBuilder::new()
-                .with_known_contracts(known_contracts)
-                .with_verbosity(verbosity);
-            // Signatures are of no value for gas reports.
-            if !self.gas_report {
-                builder = builder.with_signature_identifier(SignaturesIdentifier::new(
-                    Config::foundry_cache_dir(),
-                    config.offline,
-                )?);
-            }
-            let mut decoder = builder.build();
+            let mut decoder = decoder.clone();
```

Comment:
Do we need this clone? I added `decoder.clear_addresses();` so we can reuse the same decoder across tests

Reply:
ah right, it seemed that we needed ownership on it for `suite_result.last_run_decoder`

overlooked that we can just set it after the loop